test/README.md

```diff
@@ -0,0 +1,95 @@
+## How To Test Verdaccio
+
+Welcome to the testing folder at Verdaccio. This document aims to help you to go throug how Verdaccio show be tested.
+
+First of all we must explain the stack, we use `jest` and other tools as `supertest` to be able test the API and `puppetter` for End to End testing.
+
+We go along with the following rules in order to be consisten with all test and will make your code review smooth and fast:
+
+* We **type** all our test. eg `const foo: number = 3;`
+* **Each test should be small as possible**: You should try the `test()` block try to test only one thing and no dependencies with other test. If test requires different steps, groupd them woth `describe()` block.
+* All `test()` **headers title** should init with `test('should test ...)`: For consistency with reporting tools, this makes more easy to match the test with the feature need to be tested.
+* **Any mock data** should be located in the `partials` folder in each section.
+* Use `yaml` for **configuration files examples** instead use JSON.
+* If you use a **file based mock storage**, should be located in the `store` folder in each section.
+* All test **MUST NOT** rely in external sources and must be able to run them **off line**.
+* Test must run in the following Operative System: Unix (Mac, Linux) and Windows (7 -> latest).
+* If you are creating mock data file based and need a clean state, use `rimraf` to remove folders.
+
+## Testing Sections
+
+Verdaccio testing is being separated in 3 sections, each of them have different setup and scope, the most important is **unit test**. All sections has custom **jest configuration files**.
```

Comment:
each of them have => each of them has
is unit => is the unit
All sections has => All sections have